doc/source/whatsnew/v1.4.1.rst

```diff
@@ -15,7 +15,7 @@ including other versions of pandas.
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
 - Regression in :meth:`Series.mask` with ``inplace=True`` and ``PeriodDtype`` and an incompatible ``other`` coercing to a common dtype instead of raising (:issue:`45546`)
--
+- Regression in :func:`.assert_frame_equal` not respecting ``check_flags=False``(:issue:`45554`)
```

Comment:
again. I don't understand fully since have not investigated but this issue is not actually a regression if the flag was not propagated before? As with `check_flags=True` I assume it did not raise if the flags were different.